torch/_inductor/codegen/cpp.py

```diff
@@ -311,6 +311,26 @@ def maximum(a, b):
     def square(a):
         return f"{a}.pow(2)"
 
+    @staticmethod
+    def sign(x):
+        code = BracesBuffer()
+        # auto tmp5 = tmp4 < 0 ? -1 : 1;
+        vec_zero = f"decltype({x})(0)"
+        vec_one = f"decltype({x})(1)"
+        vec_minus_one = f"decltype({x})(-1)"
```

Comment:
Check ATen's implementation which saves one register than this:
```c++
    AT_DISPATCH_ALL_TYPES_AND2(kBFloat16, ScalarType::Half, iter.dtype(), "sign_cpu", [&]() {
        auto zero_vec = Vectorized<scalar_t>(static_cast<scalar_t>(0));
        auto one_vec = Vectorized<scalar_t>(static_cast<scalar_t>(1));

        cpu_kernel_vec(
          iter,
          [=](scalar_t a) -> scalar_t { return (0 < a) - c10::is_negative(a); },
          [=](Vectorized<scalar_t> self_vec){

              // Comparison operators returns bitmask.
              auto left = Vectorized<scalar_t>::blendv(zero_vec, one_vec, zero_vec < self_vec);
              auto right = Vectorized<scalar_t>::blendv(zero_vec, one_vec, self_vec < zero_vec);

              return left - right;
          });
    });
```

Reply:
Compare with the current implementation, aten implementation introduces `sub` computation. Current implementation only calls two `blendv`.
```C++
auto tmp5 = tmp4 < 0 ? -1 : 1;
auto tmp6 = tmp4 == 0 ? 0 : tmp5;
```

Reply:
I know but I'm not sure what is more critical to performance. My guess is less register usage since instruction-level parallel can mitigate one more `sub`?